src/app/modules/main/browser_section/bookmark/module.nim

```diff
@@ -0,0 +1,77 @@
+import NimQml
+import io_interface
+import ../io_interface as delegate_interface
+import item
+import view
+import controller
+import ../../../../global/global_singleton
+import ../../../../../app_service/service/bookmarks/service as bookmark_service
+import ../../../../core/eventemitter
+
+export io_interface
+
+type
+  Module* = ref object of io_interface.AccessInterface
+    delegate: delegate_interface.AccessInterface
+    view: View
+    viewVariant: QVariant
+    moduleLoaded: bool
+    bookmarksLoaded: bool
+    controller: Controller
+
+proc newModule*(delegate: delegate_interface.AccessInterface, events: EventEmitter, bookmarkService: bookmark_service.Service): Module =
+  result = Module()
+  result.delegate = delegate
+  result.view = view.newView(result)
+  result.viewVariant = newQVariant(result.view)
+  result.moduleLoaded = false
+  result.bookmarksLoaded = false
+  result.controller = controller.newController(result, events, bookmarkService)
+
+method delete*(self: Module) =
+  self.view.delete
+  self.viewVariant.delete
+  self.controller.delete
+
+method load*(self: Module) =
+  singletonInstance.engine.setRootContextProperty("bookmarkModule", self.viewVariant)
+  self.controller.init()
+  self.view.load()
+
+method onActivated*(self: Module) =
+  if self.bookmarksLoaded:
+    return
+
+  let bookmarks = self.controller.getBookmarks()
+  for b in bookmarks:
+    self.view.addItem(initItem(b.name, b.url, b.imageUrl))
```

Comment:
We could improve that and add all items in one call

Reply:
https://github.com/status-im/status-desktop/issues/18556